### PR TITLE
build: clean up and test auto release

### PR DESCRIPTION
**Pull request by ktmud**
🏆 Enhancements

Some minor changes to test auto-release setup by @kristw .

Reading from the code, it seems we are only [bumping versions](https://github.com/lerna/lerna/tree/master/commands/publish#positionals) from manually updated `package.json`.

1. A minor fix to speed up commit hook
2. A minor change in BigNumber to test auto release
3. Clean up some irrelevant code that is not used in the current auto-release process.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -54,34 +54,17 @@ jobs:
...
-        git update-index --assume-unchanged .npmrc
-        git tag -d `git tag | grep -E '^trigger-'`
-        yarn ci:release-from-tag
-      env:
-        NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GH_TOKEN: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+        echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" >> $HOME/.npmrc 2> /dev/null
+        npm whoami
+        yarn ci:release-from-tag  # lerna publish from-package
```
Comment (kristw) [inline .github/workflows/release.yml:70]: Please do not change these yet. 
`yarn ci:release-from-tag` actually did not bump the package version, only publish to npm by comparing the versions in `package.json` and the latest version on `npm`.

I had called a command that actually bump the package version called before, but removed that temporarily due to conflict with protected `master` branch and I didn't have time to fix and test. These credentials that you have removed in the PR are necessary for the version bump, git push and create github release to succeed.

Comment (ktmud) [reply .github/workflows/release.yml:70]: Thought we'd be manually update package versions and only use `lerna publish from-package`. If we are going to automatically update versions and make commits in CI, should we publish with `--canary` instead?

Comment (ktmud) [reply .github/workflows/release.yml:70]: I sense it'd be quite troublesome if we let CI automatically make new commits. It'd basically mean every merge will create another commit in Git.

I like what [zeit/now is doing](https://github.com/zeit/now/blob/master/utils/publish.sh): publish stable versions from git tags, but canary versions from pushes to the master branch.

Comment (kristw) [reply .github/workflows/release.yml:70]: Canary sounds good. We can discuss more. I was hesitating about publishing every version as well so I pulled it out.

**Comment by ktmud** [comment]
This is no longer needed for now since `yarn ci:create-patch-version` seems to work quite well.

**Pull request closed without merging**